backend/app.py

```diff
@@ -149,49 +192,93 @@ async def cast_vote(vote: Vote, db_engine: AbstractEngine = Depends(get_db_engin
 )
-async def read_parties_by_game(game_id: int, db_engine: AbstractEngine = Depends(get_db_engine)) -> list[Party]:
-    return db_engine.get_parties(game_id=game_id)
+async def read_parties_by_game(
+    game_id: int, db_engine: AbstractEngine = Depends(get_db_engine)
+) -> list[Party]:
+    # DEBUG: log raw SQLModel query here
+    from sqlmodel import select
+    from database.sql import models as m
+    from sqlmodel import Session
+
+    with Session(get_db_engine().engine) as sess:
+        stmt = (
+            select(m.Party)
+            .join(m.Round, m.Party.round_id == m.Round.id)
+            .where(m.Round.game_id == game_id)
+        )
+        parties_list = sess.exec(stmt).all()
+        print(
+            f"[DEBUG] Queried in endpoint, found {len(parties_list)} parties:",
+            parties_list,
+        )
+
+    # now call your real engine
+    try:
+        return db_engine.get_parties(game_id=game_id)
+    except NoDataFoundError:
+        print(f"[DEBUG] NoDataFoundError for game {game_id}")
+        return []
+
 
 @common_router.get(
     "/game/{game_id}",
-    response_model=Party,
+    response_model=Game,
 )
-async def read_game(game_id: int) -> Game:
-    return db_engine.get_game(game_id)
+async def read_game(
+    game_id: int,
+    db_engine: AbstractEngine = Depends(get_db_engine),
+) -> Game:
+    return db_engine.get_game(game_id=game_id)
 
 
 @common_router.get(
     "/game/{game_id}/rounds",
     response_model=list[Round],
 )
-async def get_rounds_by_game(game_id: int, db_engine: AbstractEngine = Depends(get_db_engine)) -> list[Round]:
+async def get_rounds_by_game(
+    game_id: int, db_engine: AbstractEngine = Depends(get_db_engine)
+) -> list[Round]:
     return db_engine.get_rounds(game_id=game_id)
 
 
 @common_router.get(
     "/join",
     response_model=Game,
 )
-async def get_game_by_hash(game_hash: str, db_engine: AbstractEngine = Depends(get_db_engine)) -> Game:
+async def get_game_by_hash(
+    game_hash: str, db_engine: AbstractEngine = Depends(get_db_engine)
+) -> Game:
     try:
         return db_engine.get_game_by_hash(game_hash=game_hash)
     except Exception:
         return Response(status_code=HTTPStatus.NOT_FOUND)
 
+
 @common_router.post(
     "/register",
     response_model=Voter,
 )
 @broadcast_game_state
-async def register_user(user: Voter, db_engine: AbstractEngine = Depends(get_db_engine)) -> Voter:
-    return db_engine.add_voter(voter=user)
-
-@common_router.post(
-    "/register_to_vote",
-    response_model=Affiliation
-)
+async def register_user(
+    user: Voter, db_engine: AbstractEngine = Depends(get_db_engine)
+) -> JSONResponse:
+    voter = db_engine.add_voter(voter=user)
+    payload = {
+        "id": voter.id,
+        "name": voter.name,
+        "game_id": voter.game_id,
+        "extra_info": voter.extra_info,
+    }
+    resp = JSONResponse(status_code=HTTPStatus.OK, content=payload)
+    resp.game_id = voter.game_id
+    return resp
```

Comment:
```suggestion
    return JSONResponse(status_code=HTTPStatus.OK, content=payload)
```
`game_id` already a part of response content + `resp.game_id = ...` will not affect the content of the response

Reply:
It is not to affect the response body content but rather the python object for the wrapper function to be able to detect correct games. There was an issue with it broadcasting multiple games to same client so I added the filtering in. Maybe there's a better way to do this but with current implementation the commented lines are needed I'm pretty sure.

Reply:
i see 🤔 
from my perspective, this is not the right way of doing this, but I have no time to figure out the right way right now 😅 
so, to not block you, can you just add a comment to the code on why this line is here